docs/lib-documentation.md

```diff
@@ -12,21 +12,22 @@ NetworkContext
 --------------
 This Kurtosis-provided class is the lowest-level representation of a test network, and provides methods for inspecting and manipulating the network. All [Network][network] implementations will encapsulate an instance of this class.
 
-### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, [ContainerConfigFactory\<S\>][containerconfigfactory] configFactory) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
+### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, [ContainerCreationConfig][containercreationconfig] containerCreationConfig, Func(String ipAddr, Map\<String, string\> generatedFileFilepaths, Map\<StaticFileID, string\> staticFileFilepaths) -\> ([ContainerRunConfig][containerrunconfig], error) generateRunConfigFunc) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
```

Comment:
Really nice job, especially with the links! Couple things to correct:
- I've been keeping a pseudo-Java style for these docs, so `Map\<String, string\>` should be `Map\<String, String\>`
- In the same vein, Java doesn't have type aliases so I've just been using `String` for the `StaticFileID` type
- `error` as a first-class type is a Go thing, so I'd just make the return type of the anonymous function `[ContainerRunConfig][containerrunconfig]` rather than `([ContainerRunConfig][containerrunconfig], error)`

Reply:
Very good explanation, thanks